packages/integration-sdk-cli/src/commands/visualize-types.ts

```diff
@@ -0,0 +1,145 @@
+import * as log from '../log';
+import * as path from 'path';
+import { createCommand } from 'commander';
+import {
+  StepGraphObjectMetadataProperties, StepEntityMetadata, StepRelationshipMetadata,
+} from '@jupiterone/integration-sdk-core';
+import { promises as fs } from 'fs';
+import { getDefaultTypesFilePath, executeTypesAction, TypesCommandArgs } from './types';
+import { generateVisHTML } from '../utils/generateVisHTML';
+import { Node, Edge } from 'vis';
+
+interface VisualizeTypesCommandArgs extends TypesCommandArgs{
+  graphFilePath: string;
+  type: string[];
+}
+
+// coercion function to collect multiple values for a flag
+const collector = (value: string, arr: string[]) => {
+  arr.push(...value.split(','));
+  return arr;
+};
+
+export function visualizeTypes() {
+  return createCommand('visualize-types')
+    .description('Generates a graph of types metadata for all steps')
+    .option(
+      '-p, --project-path <directory>',
+      'Absolute file path to the integration project directory. Defaults to the current working directory.',
+      process.cwd(),
+    )
+    .option(
+      '-g, --graph-file-path <path>',
+      'Absolute file path to the HTML file that should be created/overwritten. Defaults to {CWD}/.j1-integration/types-graph/index.html.',
+    )
+    .option(
+      '-t, --types-file-path <path>',
+      'Absolute file path to the JSON file that should be created/overwritten. Defaults to {CWD}/docs/types.json.',    
+    )
+    .option(
+      '-y, --type <string>',
+      'J1 type(s) to visualize, comma separated if multiple',
+      collector,
+      [],
+    )
+    .action(executeVisualizeTypesAction);
+}
+
+async function executeVisualizeTypesAction(
+  options: VisualizeTypesCommandArgs,
+): Promise<void> {
+  const { projectPath } = options;
+  const types = options.type.length ? undefined : options.type;
+  const graphFilePath =
+    options.graphFilePath ||
+    getDefaultTypesGraphFilePath(projectPath);
+  const typesFilePath = 
+    options.typesFilePath || 
+    getDefaultTypesFilePath(projectPath);
+
+  await executeTypesAction({
+    projectPath,
+    typesFilePath,
+  });
+  
+  log.info('\nLoading types metadata file...\n');
+  const metadataString = await fs.readFile(typesFilePath, {
+    encoding: 'utf-8',
+  });
+
+  const metadata: StepGraphObjectMetadataProperties = JSON.parse(metadataString);
+
+  if (!metadata.entities.length && !metadata.relationships.length) {
+    log.info(
+      'No entities or relationships found to generate types graph for. Exiting.',
+    );
+    return;
+  }
+
+  log.info('\nGenerating local graph from types metadata...\n');
+
+  const edges = getEdgesFromStepRelationshipMetadata(metadata.relationships, { types });
+  const nodes = getNodesFromStepEntityMetadata(metadata.entities, { types, edges });
+  
+
+  const visHtml = generateVisHTML(
+    nodes,
+    edges,
+  )
```

Comment:
iirc, the other cli command has a snapshot test. It would be nice to do the same here and also unit test the construction of the metadata graph.